File: 690/python3/main.py
"""
# Definition for Employee.
class Employee:
    def __init__(self, id: int, importance: int, subordinates: List[int]):
        self.id = id
        self.importance = importance
        self.subordinates = subordinates
"""

class Solution:
    def dfs(self, id: int, employeeMap:dict) -> int:
        e = employeeMap[id]
        result  = e.importance

        for subordinate in e.subordinates:
            result += self.dfs(subordinate, employeeMap)

        return result

    def getImportance(self, employees: list['Employee'], id: int) -> int:
        # 构建map
        employeeMap = {e.id: e for e in employees}
        return self.dfs(id, employeeMap)

File: 690/go/main.go
/**
 * Definition for Employee.
 * type Employee struct {
 *     Id int
 *     Importance int
 *     Subordinates []int
 * }
 */
type Employee struct {
    Id int
    Importance int
    Subordinates []int
}

func dfs(id int, eMap map[int]*Employee) int {
    e := eMap[id]
    result := e.Importance
    for _, subId := range e.Subordinates {
        result += dfs(subId, eMap)
    }
    return result
}

func getImportance(employees []*Employee, id int) int {
    eMap := make(map[int]*Employee)

    for _, e := range employees {
        eMap[e.Id] = e
    }

    return dfs(id, eMap)
}
